client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -95,6 +109,22 @@ export const useStepNavigationWithTracking = ( {
 					stepNavigation.exitFlow?.( to );
 				},
 			} ),
+			/**
+			 * If the `previousStep` is defined in the store, it's a solid proxy to guess that we navigated at least once via Stepper's React Router.
+			 * If the flow doesn't define a `goBack` handler, and `previousStep` is defined, we can just go history.back() and we'll remain in the flow.
+			 * But if `previousStep` is not defined, and the flow doesn't define a `goBack` handler, we should return undefined so the StepContainer doesn't render a back button.
+			 */
+			...( canUserGoBack && {
```

Comment:
@alshakero Introducing this synthetic `goBack` function causes a brief flicker in the logged out `onboarding` flow when the user transitions from the signup step to the domains step.

Reply:
I think is a fix. Improves the `canUserGoBack` logic. https://github.com/Automattic/wp-calypso/pull/102022